services/motion/builtin/moveRequest.go

```diff
@@ -0,0 +1,203 @@
+			}
+			kinematicsOptions.PositionOnlyMode = motionProfile == motionplan.PositionOnlyMotionProfile
+		}
+	}
+	ms.logger.Debugf("base limits: %v", limits)
+
+	// create a KinematicBase from the componentName
+	baseComponent, ok := ms.components[componentName]
+	if !ok {
+		return nil, resource.NewNotFoundError(componentName)
+	}
+	b, ok := baseComponent.(base.Base)
+	if !ok {
+		return nil, fmt.Errorf("cannot move component of type %T because it is not a Base", baseComponent)
+	}
+
+	kb, err := kinematicbase.WrapWithKinematics(ctx, b, ms.logger, localizer, limits, kinematicsOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	// create a new empty framesystem which we add the kinematic base to
+	fs := referenceframe.NewEmptyFrameSystem("")
+	kbf := kb.Kinematics()
+	if err := fs.AddFrame(kbf, fs.World()); err != nil {
+		return nil, err
+	}
+
+	// TODO(RSDK-3407): this does not adequately account for geometries right now since it is a transformation after the fact.
+	// This is probably acceptable for the time being, but long term the construction of the frame system for the kinematic base should
+	// be moved under the purview of the kinematic base wrapper instead of being done here.
+	offsetFrame, err := referenceframe.NewStaticFrame("offset", movementSensorToBase.Pose())
+	if err != nil {
+		return nil, err
+	}
+	if err := fs.AddFrame(offsetFrame, kbf); err != nil {
+		return nil, err
+	}
+
+	return &moveRequest{
+		planRequest: &motionplan.PlanRequest{
+			Logger:             ms.logger,
+			Goal:               referenceframe.NewPoseInFrame(referenceframe.World, goal),
+			Frame:              offsetFrame,
+			FrameSystem:        fs,
+			StartConfiguration: referenceframe.StartPositions(fs),
+			WorldState:         worldState,
+			Options:            extra,
+		},
+		actuator: kb,
+		execute: func(ctx context.Context, plan motionplan.Plan) moveResponse {
+			waypoints, err := plan.GetFrameSteps(kb.Name().Name)
+			if err != nil {
+				return moveResponse{err: err}
+			}
+
+			for i := 1; i < len(waypoints); i++ {
+				select {
+				case <-ctx.Done():
+					return moveResponse{}
+				default:
+					ms.logger.Info(waypoints[i])
+					if err := kb.GoToInputs(ctx, waypoints[i]); err != nil {
+						// If there is an error on GoToInputs, stop the component if possible before returning the error
+						if stopErr := kb.Stop(ctx, nil); stopErr != nil {
+							return moveResponse{err: errors.Wrap(err, stopErr.Error())}
+						}
+						return moveResponse{err: err}
+					}
+				}
+			}
+
+			// the plan has been fully executed so check to see if the GeoPoint we are at is close enough to the goal.
+			position, _, err := movementSensor.Position(ctx, nil)
+			if err != nil {
+				return moveResponse{err: err}
+			}
+			if spatialmath.GeoPointToPose(position, destination).Point().Norm() <= motionCfg.PlanDeviationMM {
+				return moveResponse{success: true}
+			}
+			return moveResponse{err: errors.New("reached end of plan but not at goal")}
```

Comment:
[nit/q] why not log this and replan?

Reply:
I think its easier to consider this an error for now.  In practice this really shouldn't happen